br/pkg/utils/backoff.go

```diff
@@ -204,3 +210,74 @@ func (bo *pdReqBackoffer) NextBackoff(err error) time.Duration {
 func (bo *pdReqBackoffer) Attempt() int {
 	return bo.attempt
 }
+
+type flashbackBackoffer struct {
+	attempt      int
+	delayTime    time.Duration
+	maxDelayTime time.Duration
+}
+
+// NewBackoffer creates a new controller regulating a truncated exponential backoff.
+func NewFlashBackBackoffer() Backoffer {
+	return &flashbackBackoffer{
+		attempt:      flashbackRetryTime,
+		delayTime:    flashbackWaitInterval,
+		maxDelayTime: flashbackMaxWaitInterval,
+	}
+}
+
+// specific retry error for flashback
+// it happened when region created without leader
+// region heartbeat sent may some delay (10s), during the prepare flashback, leader may not sent heartbeat yet.
+var retryableError = []string{
+	"receive regions with no peer",
+}
+
+// messageIsRetryable checks whether the message returning from TiKV client-go.
+func messageIsRetryable(msg string) bool {
+	msgLower := strings.ToLower(msg)
+	// UNSAFE! TODO: Add a error type for retryable connection error. It may need a refactor among tikv, client-go and tidb.
+	for _, errStr := range retryableError {
+		if strings.Contains(msgLower, errStr) {
+			return true
+		}
+	}
+	return false
+}
+
+func (bo *flashbackBackoffer) NextBackoff(err error) time.Duration {
+	if messageIsRetryable(err.Error()) {
+		bo.delayTime = 2 * bo.delayTime
+		bo.attempt--
+	} else {
+		e := errors.Cause(err)
+		switch e { // nolint:errorlint
+		case berrors.ErrKVEpochNotMatch, berrors.ErrPDLeaderNotFound:
```

Comment:
When will these errors happen?

Reply:
remove them?

Reply:
I think so

Reply:
fixed in https://github.com/pingcap/tidb/pull/41059/commits/c1626655d35c8d113ccf9bde3914caff0ccd3b64